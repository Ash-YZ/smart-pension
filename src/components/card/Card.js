import React from 'react'
import PropTypes from 'prop-types'
import './styles.scss'

const Card = props => {
  return (
    <div className={`card ${props.isWinner ? 'winner' : ''} ${props.isDraw ? 'draw' : ''}`}>
      <div className="card-body">
        <h5 className="card-title">{props.card ? props.card.name : '-----'}</h5>
        {props.gameMode === 'people' ?
          <p className="card-text">
            <span className={`${props.isWinner ? 'badge badge-light' : ''}`}>
              Mass: {props.card ? props.card.mass : '-----'}
            </span>
          </p> :
          <p className="card-text">
            <span className={`${props.isWinner ? 'badge badge-light' : ''}`}>
              Crew: {props.card ? props.card.crew : '-----'}
            </span>
          </p>}
      </div>
    </div>
  )
}

Card.propTypes = {
  gameMode: PropTypes.string,
  card: PropTypes.shape({
    name: PropTypes.string,
    mass: PropTypes.string,
    crew: PropTypes.string,
  }),
  isWinner: PropTypes.bool,
  isDraw: PropTypes.bool
}

export default Card;