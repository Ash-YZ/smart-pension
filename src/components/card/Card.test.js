import React from 'react';
import { shallow } from 'enzyme'
import Card from './Card'

const testProps = {
  gameMode: '',
  card: {
    name: 'Person 1',
    mass: '100'
  },
}

describe('Card', () => {
  it('should render "people" mode correctly ', () => {
    testProps.gameMode = 'people'
    const component = shallow(<Card {...testProps} />)
    expect(component).toMatchSnapshot()
  })

  it('should render "starship" mode correctly ', () => {
    testProps.gameMode = 'starship'
    const component = shallow(<Card {...testProps} />)
    expect(component).toMatchSnapshot()
  })

  it('should render empty card correctly ', () => {
    testProps.gameMode = 'people'
    testProps.card = null;
    const component = shallow(<Card {...testProps} />)
    expect(component).toMatchSnapshot()
  })
})