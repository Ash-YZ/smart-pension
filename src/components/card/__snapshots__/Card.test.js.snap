// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render "people" mode correctly  1`] = `
<div
  className="card  "
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Person 1
    </h5>
    <p
      className="card-text"
    >
      <span
        className=""
      >
        Mass: 
        100
      </span>
    </p>
  </div>
</div>
`;

exports[`Card should render "starship" mode correctly  1`] = `
<div
  className="card  "
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Person 1
    </h5>
    <p
      className="card-text"
    >
      <span
        className=""
      >
        Crew: 
      </span>
    </p>
  </div>
</div>
`;

exports[`Card should render empty card correctly  1`] = `
<div
  className="card  "
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      -----
    </h5>
    <p
      className="card-text"
    >
      <span
        className=""
      >
        Mass: 
        -----
      </span>
    </p>
  </div>
</div>
`;
